solutions/AB.js

```diff
@@ -0,0 +1,28 @@
+const fs = require("fs");
+const filePath = process.platform === "linux" ? "/dev/stdin" : INPUT_PATH;
+const input = fs.readFileSync(filePath).toString().trim().split("\n");
+
+function solution(inputArguments) {
+  const [A, B] = inputArguments.split(" ").map(Number);
+  const queue = [[B, 1]];
+
+  while (queue.length > 0) {
+    const [current, count] = queue.shift();
+
+    if (current === A) {
+      return count;
+    }
+
+    if (current % 2 === 0) {
+      queue.push([current / 2, count + 1]);
+    }
+
+    if (current % 10 === 1) {
+      queue.push([Math.floor(current / 10), count + 1]);
+    }
+  }
+
+  return -1;
+}
```

Comment:
```suggestion
  while (queue.length > 0) {
    const [current, count] = queue.shift();

    if (current === A) {
      return count;
    }
    
    if (current < A) {
      break;
    }

    if (current % 2 === 0) {
      queue.push([current / 2, count + 1]);
    }

    if (current % 10 === 1) {
      queue.push([Math.floor(current / 10), count + 1]);
    }
  }

  return -1;
}
```

`3 8`이 입력으로 들어올 경우 무한루프가 도네요!

만약, `current`가  `A`보다 작은 경우에는 그 이상 진행할 필요가 없으니, `break`를 걸어주면 해결 될 것 같습니다!